src/sentry/static/sentry/app/components/accordion.jsx

```diff
@@ -0,0 +1,83 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+import {Flex} from 'grid-emotion';
+import Badge from 'app/components/badge';
+import Button from 'app/components/buttons/button';
+
+class AccordionButton extends React.Component {
+  static propTypes = {
+    labelClosed: PropTypes.string.isRequired,
+    labelOpen: PropTypes.string,
+    cutoff: PropTypes.number,
+    children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  };
+
+  static defaultProps = {
+    cutoff: 0,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  handleClick = e => {
+    e.preventDefault();
+    this.setState({isOpen: !this.state.isOpen});
+  };
+
+  render() {
+    const {labelOpen, labelClosed, cutoff} = this.props;
+    const children = React.Children.toArray(this.props.children);
+
+    if (React.Children.count(children) <= cutoff + 1) return children;
+
+    return (
+      <div>
+        {children.slice(0, cutoff)}
+        <StyledButton onClick={this.handleClick} open={this.state.isOpen}>
+          <Flex align="center" justify="space-between" w={1}>
+            <Flex align="center">
+              {this.state.isOpen ? labelOpen || labelClosed : labelClosed}
+              <Badge style={{marginLeft: space(0.25)}} text={children.length - cutoff} />
+            </Flex>
+            <StyledInlineSvg src="icon-chevron-down" open={this.state.isOpen} />
+          </Flex>
+        </StyledButton>
+        {this.state.isOpen && children.slice(cutoff)}
+      </div>
+    );
+  }
+}
+
+const StyledButton = styled(Button)`
+  display: block;
+  width: 100%;
```

Comment:
is this width needed since it's display block?

Reply:
The `button` element is weird (like images, textareas, etc.) and won't auto-fill it's parent width even when set to `display: block`. 

I could do the opposite and remove `display: block` but then the button gets a line-height which effects the clarity of the code (I would need to set `line-height: 0` OR the harmony for the margins (it's hard to set equal margins on top and bottom when working w/ line height).